compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -1026,12 +984,15 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         move_span: Span,
         move_spans: UseSpans<'tcx>,
         moved_place: Place<'tcx>,
-        partially_str: &str,
-        loop_message: &str,
-        move_msg: &str,
+        is_partial: bool,
+        is_loop_message: bool,
+        is_move_msg: bool,
         is_loop_move: bool,
         maybe_reinitialized_locations_is_empty: bool,
     ) {
+        use crate::session_diagnostics::CaptureReasonLabel::*;
+        use crate::session_diagnostics::CaptureReasonNote::*;
+        use crate::session_diagnostics::CaptureReasonSuggest::*;
```

Comment:
I think if we import the types at the top of the file then adding the variants to function/closure scopes won't be necessary.